website/docs/stitch-type-merging.md

```diff
@@ -146,17 +144,28 @@ const postsSchema = makeExecutableSchema({
 });
 ```
 
-In this example, `userById` simply converts the submitted ID into stub record that gets resolved as the local `User` type.
+In this example, the `postUserById` resolver simply converts a submitted user ID into stub record that gets resolved as the local `User` type.
+
+### Null records
+
+The above example will always resolve a stubbed `User` record for _any_ requested ID. For example, requesting ID `7` (which has no associated posts) would return as:
+
+```js
+{ id: '7', posts: [] }
+```
+
+This fabricated result fulfills the not-null requirement of the `posts:[Post]!` field. However, it also makes the posts service awkwardly responsible for data known only by omission. A cleaner solution would be to loosen schema nullability down to `posts:[Post]`, and then return `null` for unknown user IDs without associated posts. Null is always a valid merge object as long as the unique fields it fulfills are nullable.
```

Comment:
This is a subtle point, at least from where I sit. Great to document... Are there other solutions also? What is the fields it returns are not nullable?

Reply:
I tried a few rewording and expansion approaches, and all led to precarious ends that became misleading or overstated. As you say – this is an incredibly subtle feature (though one of the best new additions in v7 IMO!)

I'd be in favor of keeping it as-is for now and letting the less-is-more approach ride until we get some more feedback. I think this is one of those sections that devs will ignore at first no matter how detailed it is, but then will revisit later once they hit a situation where it applies, at which time the explanation makes perfect sense. Being too direct up front can lead to the inverse problem of making it sound more complex than it really is.